File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int na=nums.size();
        unordered_map<int,int> numMap;
        for(int i=0;i<na;i++){
            int rem=target-nums[i];
            if(numMap.count(rem)){
                return {numMap[rem],i};
            }
            numMap[nums[i]]=i;
        }
        return {};
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, li: list[int], target: int) -> list[int]:
        seen={}
        for i in range(len(li)):
            diff=target-li[i]
            if diff in seen:
                return [seen[diff],i]
            seen[li[i]]=i
        return []